optuna/integration/skopt.py

```diff
@@ -177,6 +177,10 @@ def sample_relative(
             return {}
 
         optimizer = _Optimizer(search_space, self._skopt_kwargs)
+
+        if "random_state" in self._skopt_kwargs:
+            self._skopt_kwargs["random_state"] += 1
```

Comment:
I'm not sure if `np.random.RandomState` can generate random sequence with the random seed updated incrementally. How about reusing the `self._optimizer.rng`?

```diff
diff --git a/optuna/integration/skopt.py b/optuna/integration/skopt.py
index 662be194d..2ffe1eba4 100644
--- a/optuna/integration/skopt.py
+++ b/optuna/integration/skopt.py
@@ -285,6 +285,7 @@ class _Optimizer:
             dimensions.append(dimension)
 
         self._optimizer = skopt.Optimizer(dimensions, **skopt_kwargs)
+        skopt_kwargs["random_state"] = self._optimizer.rng
 
     def tell(self, study: Study, complete_trials: List[FrozenTrial]) -> None:
```

Reply:
I updated to reuse the random state. PTAL.

Reply:
Thank you for your update. Your [change](https://github.com/optuna/optuna/pull/4023/commits/f54e1ef4278d3a27dc8734f612e011a3850ef5f1)  looks solid.